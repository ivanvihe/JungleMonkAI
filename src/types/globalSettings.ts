import type { AgentManifestCache } from './agents';

export type BuiltinProvider = 'openai' | 'anthropic' | 'groq' | 'github' | 'gitlab';

export type SupportedProvider = BuiltinProvider | (string & {});

export const BUILTIN_PROVIDERS: BuiltinProvider[] = ['openai', 'anthropic', 'groq', 'github', 'gitlab'];

export type ApiKeySettings = Record<string, string>;

export interface CommandPresetSettings {
  temperature?: number;
  maxTokens?: number;
}

export interface CommandPreset {
  id: string;
  label: string;
  prompt: string;
  description?: string;
  provider?: string;
  model?: string;
  settings?: CommandPresetSettings;
}

export interface RoutingRule {
  provider: string;
  model: string;
  commandPresetId?: string;
}

export type DefaultRoutingRules = Record<string, RoutingRule>;

export interface GlobalSettings {
  version: number;
  apiKeys: ApiKeySettings;
  commandPresets: CommandPreset[];
  defaultRoutingRules: DefaultRoutingRules;
  enabledPlugins: string[];
  approvedManifests: AgentManifestCache;
}
