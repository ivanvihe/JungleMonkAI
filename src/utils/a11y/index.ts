export * from './focus';
export * from './aria';
