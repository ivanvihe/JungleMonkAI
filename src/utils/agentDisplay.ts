import { AgentDefinition } from '../core/agents/agentRegistry';

export const getAgentDisplayName = (agent: AgentDefinition): string =>
  agent.kind === 'local' ? 'Jarvis' : agent.name;

export const getAgentVersionLabel = (agent: AgentDefinition): string =>
  agent.kind === 'local' ? agent.name : agent.name;

export const getAgentChannelLabel = (agent: AgentDefinition): string => {
  if (agent.kind === 'local') {
    return 'Jarvis';
  }

  if (agent.channel === 'gpt') {
    return 'GPT';
  }

  if (agent.channel === 'claude') {
    return 'Claude';
  }

  if (agent.channel === 'groq') {
    return 'Groq';
  }

  return agent.name;
};
