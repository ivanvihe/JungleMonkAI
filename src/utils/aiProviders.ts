export interface ChatProviderRequest {
  apiKey: string;
  model: string;
  prompt: string;
  systemPrompt?: string;
  maxTokens?: number;
  temperature?: number;
}

const DEFAULT_MAX_TOKENS = 1024;
const DEFAULT_TEMPERATURE = 0.7;

const resolveMessage = (value: unknown): string => {
  if (!value) {
    return '';
  }
  if (typeof value === 'string') {
    return value;
  }
  if (Array.isArray(value)) {
    return value
      .map(entry => {
        if (!entry) {
          return '';
        }
        if (typeof entry === 'string') {
          return entry;
        }
        if (typeof entry === 'object' && 'text' in entry) {
          const text = (entry as { text?: string }).text;
          return typeof text === 'string' ? text : '';
        }
        return '';
      })
      .filter(Boolean)
      .join('\n');
  }
  if (typeof value === 'object' && 'content' in (value as Record<string, unknown>)) {
    return resolveMessage((value as Record<string, unknown>).content);
  }
  return '';
};

const ensureContent = (raw: string, provider: string): string => {
  const content = raw.trim();
  if (!content) {
    throw new Error(`${provider} devolvió una respuesta vacía`);
  }
  return content;
};

export const callOpenAIChat = async ({
  apiKey,
  model,
  prompt,
  systemPrompt,
  maxTokens = DEFAULT_MAX_TOKENS,
  temperature = DEFAULT_TEMPERATURE,
}: ChatProviderRequest): Promise<string> => {
  const response = await fetch('https://api.openai.com/v1/chat/completions', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${apiKey}`,
    },
    body: JSON.stringify({
      model,
      max_tokens: maxTokens,
      temperature,
      messages: [
        ...(systemPrompt
          ? [
              {
                role: 'system',
                content: systemPrompt,
              },
            ]
          : []),
        {
          role: 'user',
          content: prompt,
        },
      ],
    }),
  });

  if (!response.ok) {
    const error = await response.json().catch(() => ({}));
    throw new Error(error?.error?.message || `OpenAI respondió con ${response.status}`);
  }

  const payload = await response.json();
  const choice = payload?.choices?.[0]?.message?.content;
  return ensureContent(resolveMessage(choice), 'OpenAI');
};

export const callGroqChat = async ({
  apiKey,
  model,
  prompt,
  systemPrompt,
  maxTokens = DEFAULT_MAX_TOKENS,
  temperature = DEFAULT_TEMPERATURE,
}: ChatProviderRequest): Promise<string> => {
  const response = await fetch('https://api.groq.com/openai/v1/chat/completions', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${apiKey}`,
    },
    body: JSON.stringify({
      model,
      max_tokens: maxTokens,
      temperature,
      messages: [
        ...(systemPrompt
          ? [
              {
                role: 'system',
                content: systemPrompt,
              },
            ]
          : []),
        {
          role: 'user',
          content: prompt,
        },
      ],
    }),
  });

  if (!response.ok) {
    const error = await response.json().catch(() => ({}));
    throw new Error(error?.error?.message || `Groq respondió con ${response.status}`);
  }

  const payload = await response.json();
  const choice = payload?.choices?.[0]?.message?.content;
  return ensureContent(resolveMessage(choice), 'Groq');
};

export const callAnthropicChat = async ({
  apiKey,
  model,
  prompt,
  systemPrompt,
  maxTokens = DEFAULT_MAX_TOKENS,
  temperature = DEFAULT_TEMPERATURE,
}: ChatProviderRequest): Promise<string> => {
  const response = await fetch('https://api.anthropic.com/v1/messages', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'x-api-key': apiKey,
      'anthropic-version': '2023-06-01',
    },
    body: JSON.stringify({
      model,
      max_tokens: maxTokens,
      temperature,
      ...(systemPrompt ? { system: systemPrompt } : {}),
      messages: [
        {
          role: 'user',
          content: prompt,
        },
      ],
    }),
  });

  if (!response.ok) {
    const error = await response.json().catch(() => ({}));
    const message = error?.error?.message || error?.error || error?.message;
    throw new Error(message || `Anthropic respondió con ${response.status}`);
  }

  const payload = await response.json();
  const content = payload?.content;
  return ensureContent(resolveMessage(content), 'Anthropic');
};
