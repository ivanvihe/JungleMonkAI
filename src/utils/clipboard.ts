export const copyToClipboard = async (value: string): Promise<void> => {
  if (typeof navigator !== 'undefined' && navigator?.clipboard?.writeText) {
    await navigator.clipboard.writeText(value);
    return;
  }

  // Fallback for environments sin API de clipboard (por ejemplo tests)
  if (typeof document === 'undefined') {
    return;
  }

  const textarea = document.createElement('textarea');
  textarea.value = value;
  textarea.setAttribute('readonly', 'true');
  textarea.style.position = 'absolute';
  textarea.style.left = '-9999px';
  document.body.appendChild(textarea);
  textarea.focus();
  textarea.select();
  document.execCommand('copy');
  document.body.removeChild(textarea);
};

export default copyToClipboard;
