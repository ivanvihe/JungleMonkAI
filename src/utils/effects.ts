export const AVAILABLE_EFFECTS = [
  'none',
  'glitch1',
  'glitch2',
  'glitch3',
  'blur',
  'distortion',
  'pixelate',
  'invert',
  'sepia',
  'noise',
  'scanlines',
  'flash'
];
