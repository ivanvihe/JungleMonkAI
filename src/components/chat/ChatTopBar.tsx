import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { AgentDefinition, AgentKind } from '../../core/agents/agentRegistry';
import { AgentPresenceSummary, AgentPresenceStatus } from '../../core/agents/presence';
import { ChatActorFilter } from '../../types/chat';
import { getAgentDisplayName } from '../../utils/agentDisplay';
import { useProjects } from '../../core/projects/ProjectContext';
import { useJarvisCore, type JarvisRuntimeStatus } from '../../core/jarvis/JarvisCoreContext';

interface ChatTopBarProps {
  agents: AgentDefinition[];
  presenceSummary: AgentPresenceSummary;
  activeAgents: number;
  totalAgents: number;
  pendingResponses: number;
  activeFilter: ChatActorFilter;
  onFilterChange: (filter: ChatActorFilter) => void;
  onRefreshPresence: () => void;
  onOpenStats: () => void;
  onOpenGlobalSettings: () => void;
  onOpenPlugins: () => void;
  onOpenMcp: () => void;
  onOpenModelManager: () => void;
  activeView: 'chat' | 'repo' | 'canvas';
  onChangeView: (view: 'chat' | 'repo' | 'canvas') => void;
}

const STATUS_LABELS: Record<AgentPresenceStatus, string> = {
  online: 'Operativo',
  offline: 'En espera',
  error: 'Con incidencias',
  loading: 'Verificando…',
};

const KIND_LABELS: Record<AgentKind, string> = {
  cloud: 'Agentes en nube',
  local: 'Agentes locales',
};

const resolveStatus = (summary: AgentPresenceSummary): AgentPresenceStatus => {
  if (summary.totals.error > 0) {
    return 'error';
  }
  if (summary.totals.online > 0) {
    return 'online';
  }
  if (summary.totals.loading > 0) {
    return 'loading';
  }
  return 'offline';
};

export const ChatTopBar: React.FC<ChatTopBarProps> = ({
  agents,
  presenceSummary,
  activeAgents,
  totalAgents,
  pendingResponses,
  activeFilter,
  onFilterChange,
  onRefreshPresence,
  onOpenStats,
  onOpenGlobalSettings,
  onOpenPlugins,
  onOpenMcp,
  onOpenModelManager,
  activeView,
  onChangeView,
}) => {
  const hasPending = pendingResponses > 0;
  const overallStatus = resolveStatus(presenceSummary);
  const { projects, activeProjectId, activeProject, selectProject } = useProjects();
  const {
    runtimeStatus,
    ensureOnline,
    uptimeMs,
    lastError,
    lastHealthMessage,
  } = useJarvisCore();
  const [isEnsuring, setEnsuring] = useState(false);
  const activeAgentsMessage = useMemo(() => {
    const base = `${activeAgents} agente${activeAgents === 1 ? '' : 's'}`;
    return `${base} coordinando la conversación`;
  }, [activeAgents]);

  const handleEnsureJarvis = useCallback(async () => {
    setEnsuring(true);
    try {
      await ensureOnline();
    } finally {
      setEnsuring(false);
    }
  }, [ensureOnline]);

  const handleProjectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    const nextId = event.target.value || null;
    selectProject(nextId);
  };

  const projectOptions = useMemo(
    () =>
      projects.map(project => ({
        id: project.id,
        label: project.name,
      })),
    [projects],
  );

  const filterOptions = useMemo(() => {
    const base: { value: ChatActorFilter; label: string }[] = [
      { value: 'all', label: 'Todos los actores' },
      { value: 'user', label: 'Usuario' },
      { value: 'system', label: 'Control Hub' },
    ];

    (['cloud', 'local'] as AgentKind[]).forEach(kind => {
      const bucket = presenceSummary.byKind[kind];
      if (bucket.total > 0) {
        base.push({ value: `kind:${kind}` as ChatActorFilter, label: KIND_LABELS[kind] });
      }
    });

    agents
      .filter(agent => agent.active)
      .forEach(agent => {
        base.push({ value: `agent:${agent.id}` as ChatActorFilter, label: getAgentDisplayName(agent) });
      });

    return base;
  }, [agents, presenceSummary]);

  const filterValue = useMemo(() => {
    if (filterOptions.some(option => option.value === activeFilter)) {
      return activeFilter;
    }
    return 'all';
  }, [activeFilter, filterOptions]);

  useEffect(() => {
    if (filterValue !== activeFilter) {
      onFilterChange('all');
    }
  }, [filterValue, activeFilter, onFilterChange]);

  const jarvisStatusLabels: Record<JarvisRuntimeStatus, string> = useMemo(
    () => ({
      offline: 'Jarvis Core desconectado',
      starting: 'Jarvis Core iniciando…',
      ready: 'Jarvis Core operativo',
      error: 'Jarvis Core con incidencias',
    }),
    [],
  );

  const jarvisTooltip = useMemo(() => {
    const base = jarvisStatusLabels[runtimeStatus];
    const detail = lastError ?? lastHealthMessage;
    if (!detail) {
      return base;
    }
    return `${base} · ${detail}`;
  }, [jarvisStatusLabels, runtimeStatus, lastError, lastHealthMessage]);

  const jarvisAriaLabel = useMemo(() => {
    const base = jarvisStatusLabels[runtimeStatus];
    if (!uptimeMs || uptimeMs <= 0) {
      return base;
    }
    const totalSeconds = Math.floor(uptimeMs / 1000);
    const hours = Math.floor(totalSeconds / 3600);
    const minutes = Math.floor((totalSeconds % 3600) / 60);
    const seconds = totalSeconds % 60;
    const uptimeDescription =
      hours > 0
        ? `${hours} hora${hours === 1 ? '' : 's'} y ${minutes} minuto${minutes === 1 ? '' : 's'}`
        : `${minutes} minuto${minutes === 1 ? '' : 's'} y ${seconds} segundo${seconds === 1 ? '' : 's'}`;
    return `${base}. Tiempo en línea: ${uptimeDescription}`;
  }, [jarvisStatusLabels, runtimeStatus, uptimeMs]);

  const uptimeLabel = useMemo(() => {
    if (!uptimeMs || uptimeMs <= 0) {
      return null;
    }
    const totalSeconds = Math.floor(uptimeMs / 1000);
    const days = Math.floor(totalSeconds / 86400);
    const hours = Math.floor((totalSeconds % 86400) / 3600);
    const minutes = Math.floor((totalSeconds % 3600) / 60);
    const seconds = totalSeconds % 60;

    if (days > 0) {
      return `↑ ${days}d ${hours.toString().padStart(2, '0')}h`;
    }
    if (hours > 0) {
      return `↑ ${hours}h ${minutes.toString().padStart(2, '0')}m`;
    }
    return `↑ ${minutes}m ${seconds.toString().padStart(2, '0')}s`;
  }, [uptimeMs]);

  return (
    <header className="chat-top-bar">
      <div className="topbar-left">
        <div className="brand-mark" aria-hidden>
          🌀
        </div>
        <div className="brand-meta">
          <span className="brand-title">JungleMonk.AI</span>
          <span className="brand-status">
            <span className={`status-dot status-${overallStatus}`} aria-hidden />
            {STATUS_LABELS[overallStatus]}
          </span>
        </div>
        <div className="mode-switcher" role="tablist" aria-label="Cambiar vista">
          <button
            type="button"
            className={activeView === 'chat' ? 'is-active' : ''}
            onClick={() => onChangeView('chat')}
            role="tab"
            aria-selected={activeView === 'chat'}
          >
            💬
          </button>
          <button
            type="button"
            className={activeView === 'repo' ? 'is-active' : ''}
            onClick={() => onChangeView('repo')}
            role="tab"
            aria-selected={activeView === 'repo'}
          >
            🗂️
          </button>
          <button
            type="button"
            className={activeView === 'canvas' ? 'is-active' : ''}
            onClick={() => onChangeView('canvas')}
            role="tab"
            aria-selected={activeView === 'canvas'}
            aria-label="Abrir canvas de código"
          >
            🧪
          </button>
        </div>
      </div>

      <div className="topbar-center">
        <div className="metric-group">
          <div className="metric-chip">
            <span className="metric-label">Activos</span>
            <span className="metric-value">
              {activeAgents}/{totalAgents}
            </span>
          </div>
          <div className={`metric-chip ${hasPending ? 'is-warning' : ''}`}>
            <span className="metric-label">Pendientes</span>
            <span className="metric-value">{pendingResponses}</span>
          </div>
          <span className="metric-caption" role="status" aria-live="polite">
            {activeAgentsMessage}
          </span>
          <button
            type="button"
            className="icon-button"
            onClick={onRefreshPresence}
            aria-label="Actualizar estado de agentes"
          >
            ↻
          </button>
          <button
            type="button"
            className="icon-button"
            onClick={onOpenStats}
            aria-label="Ver estadísticas de la conversación"
          >
            📊
          </button>
          <button
            type="button"
            className={`icon-button jarvis-runtime-button status-${runtimeStatus}${
              isEnsuring ? ' is-busy' : ''
            }`}
            onClick={handleEnsureJarvis}
            aria-label={jarvisAriaLabel}
            title={jarvisTooltip}
          >
            <span className="jarvis-runtime-icon" aria-hidden>
              🤖
            </span>
            <span className="jarvis-runtime-copy" aria-hidden>
              <span className="jarvis-runtime-status">{jarvisStatusLabels[runtimeStatus]}</span>
              {uptimeLabel && <span className="jarvis-runtime-uptime">{uptimeLabel}</span>}
            </span>
          </button>
          <button
            type="button"
            className="icon-button"
            onClick={onOpenModelManager}
            aria-label="Abrir gestor de modelos"
          >
            💾
          </button>
        </div>

        <div className="project-pill">
          <select
            aria-label="Seleccionar proyecto activo"
            value={activeProjectId ?? ''}
            onChange={handleProjectChange}
            disabled={!projectOptions.length}
          >
            {projectOptions.map(option => (
              <option key={option.id} value={option.id}>
                {option.label}
              </option>
            ))}
            {!projectOptions.length && <option value="">Sin proyectos configurados</option>}
          </select>
          <div className="project-meta" aria-live="polite">
            {activeProject
              ? `${activeProject.repositoryPath}${
                  activeProject.defaultBranch ? `@${activeProject.defaultBranch}` : ''
                }`
              : 'Sin proyecto activo'}
          </div>
        </div>

        <div className="filter-pill">
          <select
            aria-label="Filtrar actores en la conversación"
            value={filterValue}
            onChange={event => onFilterChange(event.target.value as ChatActorFilter)}
          >
            {filterOptions.map(option => (
              <option key={option.value} value={option.value}>
                {option.label}
              </option>
            ))}
          </select>
        </div>
      </div>

      <div className="topbar-actions">
        <button
          type="button"
          className="icon-button"
          onClick={onOpenPlugins}
          aria-label="Abrir plugins"
        >
          🧩
        </button>
        <button type="button" className="icon-button" onClick={onOpenMcp} aria-label="Abrir perfiles MCP">
          🛰️
        </button>
        <button
          type="button"
          className="icon-button"
          onClick={onOpenGlobalSettings}
          aria-label="Ajustes globales"
        >
          ⚙️
        </button>
      </div>
    </header>
  );
};
