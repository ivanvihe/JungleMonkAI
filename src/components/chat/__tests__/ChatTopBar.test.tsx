import React from 'react';
import { describe, it, expect, afterEach, beforeEach, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { ChatTopBar } from '../ChatTopBar';
import type { AgentDefinition } from '../../../core/agents/agentRegistry';
import type { AgentPresenceSummary } from '../../../core/agents/presence';
import type { ChatActorFilter } from '../../../types/chat';
import type { JarvisCoreContextValue, JarvisRuntimeStatus } from '../../../core/jarvis/JarvisCoreContext';
import { DEFAULT_GLOBAL_SETTINGS } from '../../../utils/globalSettings';

const useJarvisCoreMock = vi.fn<JarvisCoreContextValue, []>();
const useProjectsMock = vi.fn();

vi.mock('../../../core/jarvis/JarvisCoreContext', async () => {
  const actual = await vi.importActual<typeof import('../../../core/jarvis/JarvisCoreContext')>(
    '../../../core/jarvis/JarvisCoreContext',
  );
  return {
    ...actual,
    useJarvisCore: () => useJarvisCoreMock(),
  };
});

vi.mock('../../../core/projects/ProjectContext', () => ({
  useProjects: () => useProjectsMock(),
}));

const basePresence: AgentPresenceSummary = {
  totals: { online: 0, offline: 0, loading: 0, error: 0 },
  byKind: {
    cloud: { total: 0, active: 0, online: 0, offline: 0, loading: 0, error: 0 },
    local: { total: 0, active: 0, online: 0, offline: 0, loading: 0, error: 0 },
  },
};

const defaultJarvisState = (overrides: Partial<JarvisCoreContextValue> = {}): JarvisCoreContextValue => ({
  connected: false,
  lastError: null,
  activeModel: null,
  downloads: {},
  models: [],
  runtimeStatus: 'offline',
  uptimeMs: null,
  config: DEFAULT_GLOBAL_SETTINGS.jarvisCore,
  baseUrl: 'http://127.0.0.1:8000',
  lastHealthMessage: null,
  ensureOnline: vi.fn(),
  refreshModels: vi.fn(),
  downloadModel: vi.fn(),
  activateModel: vi.fn(),
  invokeChat: vi.fn(),
  launchAction: vi.fn(),
  ...overrides,
});

const renderComponent = (props?: Partial<{ filter: ChatActorFilter; presence: AgentPresenceSummary }>) => {
  return render(
    <ChatTopBar
      agents={[] as AgentDefinition[]}
      presenceSummary={props?.presence ?? basePresence}
      activeAgents={0}
      totalAgents={0}
      pendingResponses={0}
      activeFilter={props?.filter ?? 'all'}
      onFilterChange={vi.fn()}
      onRefreshPresence={vi.fn()}
      onOpenStats={vi.fn()}
      onOpenGlobalSettings={vi.fn()}
      onOpenPlugins={vi.fn()}
      onOpenMcp={vi.fn()}
      onOpenModelManager={vi.fn()}
      activeView="chat"
      onChangeView={vi.fn()}
    />,
  );
};

describe('ChatTopBar', () => {
  beforeEach(() => {
    useProjectsMock.mockReturnValue({
      projects: [],
      activeProjectId: null,
      activeProject: null,
      selectProject: vi.fn(),
    });
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('permite relanzar la conexión de Jarvis Core', async () => {
    const ensureOnline = vi.fn().mockResolvedValue(undefined);
    useJarvisCoreMock.mockReturnValue(defaultJarvisState({ ensureOnline }));

    renderComponent();

    const jarvisButton = screen.getByRole('button', { name: /Jarvis Core desconectado/i });
    fireEvent.click(jarvisButton);

    expect(ensureOnline).toHaveBeenCalled();
  });

  const jarvisStates: Array<{ status: JarvisRuntimeStatus; uptimeMs: number | null; snapshot: string }>
    = [
      { status: 'ready', uptimeMs: 90 * 60 * 1000, snapshot: 'ready-state' },
      { status: 'starting', uptimeMs: null, snapshot: 'starting-state' },
      { status: 'error', uptimeMs: 5 * 1000, snapshot: 'error-state' },
    ];

  it.each(jarvisStates)(
    'renderiza el estado visual de Jarvis Core: %s',
    ({ status, uptimeMs, snapshot }) => {
      useJarvisCoreMock.mockReturnValue(
        defaultJarvisState({ runtimeStatus: status, uptimeMs, lastError: status === 'error' ? 'Fallo' : null }),
      );

      const { container } = renderComponent();
      const runtimeButton = container.querySelector('.jarvis-runtime-button');
      expect(runtimeButton).toBeInTheDocument();
      expect(runtimeButton).toMatchSnapshot(snapshot);
    },
  );

  it('muestra el contador de actividad cuando está disponible', () => {
    useJarvisCoreMock.mockReturnValue(defaultJarvisState({ runtimeStatus: 'ready', uptimeMs: 3_650_000 }));

    renderComponent();

    expect(screen.getByText(/↑ 1h 00m/i)).toBeInTheDocument();
  });
});
