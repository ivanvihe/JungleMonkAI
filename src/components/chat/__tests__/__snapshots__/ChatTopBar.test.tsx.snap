// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatTopBar > renderiza el estado visual de Jarvis Core: { status: 'error', uptimeMs: 5000, snapshot: 'error-state' } > error-state 1`] = `
<button
  aria-label="Jarvis Core con incidencias. Tiempo en lÃ­nea: 0 minutos y 5 segundos"
  class="icon-button jarvis-runtime-button status-error"
  title="Jarvis Core con incidencias Â· Fallo"
  type="button"
>
  <span
    aria-hidden="true"
    class="jarvis-runtime-icon"
  >
    ðŸ¤–
  </span>
  <span
    aria-hidden="true"
    class="jarvis-runtime-copy"
  >
    <span
      class="jarvis-runtime-status"
    >
      Jarvis Core con incidencias
    </span>
    <span
      class="jarvis-runtime-uptime"
    >
      â†‘ 0m 05s
    </span>
  </span>
</button>
`;

exports[`ChatTopBar > renderiza el estado visual de Jarvis Core: { status: 'ready', uptimeMs: 5400000, snapshot: 'ready-state' } > ready-state 1`] = `
<button
  aria-label="Jarvis Core operativo. Tiempo en lÃ­nea: 1 hora y 30 minutos"
  class="icon-button jarvis-runtime-button status-ready"
  title="Jarvis Core operativo"
  type="button"
>
  <span
    aria-hidden="true"
    class="jarvis-runtime-icon"
  >
    ðŸ¤–
  </span>
  <span
    aria-hidden="true"
    class="jarvis-runtime-copy"
  >
    <span
      class="jarvis-runtime-status"
    >
      Jarvis Core operativo
    </span>
    <span
      class="jarvis-runtime-uptime"
    >
      â†‘ 1h 30m
    </span>
  </span>
</button>
`;

exports[`ChatTopBar > renderiza el estado visual de Jarvis Core: { status: 'starting', uptimeMs: null, snapshot: 'starting-state' } > starting-state 1`] = `
<button
  aria-label="Jarvis Core iniciandoâ€¦"
  class="icon-button jarvis-runtime-button status-starting"
  title="Jarvis Core iniciandoâ€¦"
  type="button"
>
  <span
    aria-hidden="true"
    class="jarvis-runtime-icon"
  >
    ðŸ¤–
  </span>
  <span
    aria-hidden="true"
    class="jarvis-runtime-copy"
  >
    <span
      class="jarvis-runtime-status"
    >
      Jarvis Core iniciandoâ€¦
    </span>
  </span>
</button>
`;
